### PR TITLE
Move online status into modal

### LINKED ISSUE #22606: Unable to open status selection when scrolled down
Also, the status select popover moves out of the viewport when scrolling and flutters while scrolling. See here:

status

This is due to the `boundaries-element` being `body` which moves out of the viewport. The flutter is due to the container being `body` as well, which moves. See https://github.com/nextcloud/nextcloud-vue/issues/1384 for details.

Server version: latest master.

### PR DESCRIPTION
**Pull request by skjnldsv**
Capture d’écran_2020-10-01_22-07-46
Capture d’écran_2020-10-01_22-12-15

### CODE REVIEW

**Comment by skjnldsv** [comment]
Fixed a small bug on the button styling.
@juliushaertl the dashboard expect an Actions component and override the styling.
I had to copy/paste it so it fits as the `.action-item__menutoggle` selector you are using doesn't apply here anymore.

What do you wanna do? Extend your selector? Leave it like that? :shrug:

**Comment by skjnldsv** [comment]
I also found a bug that is unrelated I think. When you submit an empty message you don't get a warning stating the message is empty, it still tries to push the empty custom status message

**Comment by jancborchardt** [comment]
Looks and works great, only the :focus doesn’t work – checking to see what’s wrong. Probably the focus is on a different element.

**Comment by skjnldsv** [comment]
> only the :focus doesn’t work – checking to see what’s wrong

The focus on what? If that is on the online radio selectors this is because there is no different design for checked/focused. Also the default browser is to use arrow keys for radio selectors. Not tab :v:

**Comment by nickvergessen** [comment]
> I also found a bug that is unrelated I think. When you submit an empty message you don't get a warning stating the message is empty, it still tries to push the empty custom status message

Fixed that earlier https://github.com/nextcloud/server/pull/23108

File: apps/user_status/src/components/OnlineStatusSelect.vue [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		padding-left: $icon-size + $label-padding * 2;
+		border: 2px solid var(--color-main-background);
+		border-radius: var(--border-radius-large);
+		background-color: var(--color-background-hover);
+		background-position: $label-padding center;
+		background-size: $icon-size;
+	}
+
+	&__input:checked + &__label,
+	&__input:focus + &__label,
```
Comment (jancborchardt) [inline apps/user_status/src/components/OnlineStatusSelect.vue:97]: This does look correct, and in the inspector when you trigger :focus it does work, but it’s not in the tab order for some reason?

Comment (skjnldsv) [reply apps/user_status/src/components/OnlineStatusSelect.vue:97]: https://github.com/nextcloud/server/pull/23130#issuecomment-702379331

**Comment by nickvergessen** [comment]
The hint about notifications is missing for DND?

**Comment by jancborchardt** [comment]
> > only the :focus doesn’t work – checking to see what’s wrong
> 
> The focus on what? If that is on the online radio selectors this is because there is no different design for checked/focused. Also the default browser is to use arrow keys for radio selectors. Not tab v

Aaaaah, keep forgetting about the arrow keys, jeez! Then all fine :D

**Review by jancborchardt — APPROVED** [review]
Tested and works very nicely!

**Comment by jancborchardt** [comment]
> The hint about notifications is missing for DND?

Yep, to fix the look then, "Invisible" could also get a subline saying "Appear offline" and then it would look nice. As discussed @skjnldsv 

Also found another issue that the modal is not scrollable. The "Customize" modal of the Dashboard is, so we should do the same adjustment here. @juliushaertl did you do anything special there?

**Comment by jancborchardt** [comment]
> > The hint about notifications is missing for DND?
> 
> Yep, to fix the look then, "Invisible" could also get a subline saying "Appear offline" and then it would look nice. As discussed @skjnldsv
> 
> Also found another issue that the modal is not scrollable. The "Customize" modal of the Dashboard is, so we should do the same adjustment here. @juliushaertl did you do anything special there?

Fixed both of those issues!

Just needs the commit from @skjnldsv to show the subline.

**Comment by nickvergessen** [comment]
Bildschirmfoto von 2020-10-02 09-32-51

Subline added

**Review by rullzer — APPROVED** [review]

**Pull request merged**